### ISSUE TITLE
Why do the fs and stream tokio readers not return tokio::read::ZipEntryReader

### ISSUE THREAD

**Issue by jplatte** [Majored/rs-async-zip#89]
Why is it that only the `tokio::read::seek` module has a `ZipFileReader` that returns `tokio::read::ZipEntryReader`s? That means that one has to wrap the entry reader(s) in `tokio_util::compat::Compat` to use them in a tokio context.

**Comment by mineichen** [comment]
The trait mustn't have a blanked implementation. But you're right, the two Generic-Implementations for tokio::AsyncRead and futures::AsyncRead could overlap and therefore conflict... Is this what you meant?

Maybe a 'ZipFileReader::new_tokio' would be a better choice

**Comment by Majored** [comment]
My only problem with having a `new_tokio()` is that we'd be returning a type where the reader is `Compat<R>`, which means if the library user wants to reference this type, they will have to pull in `tokio-util` themselves (which I want to avoid if possible).

That's why my current start on converting them over use types that shadow the base implementation but wrapping `Compat<R>` and the calls to `.compat()`internally to avoid having to pull it in.

**Comment by mineichen** [comment]
True, that import would be bad for ergonomics. We could however just export a Type without the need for a Wrapper:

´pub type TokioZipFileReader<T> = ZipFileReader<tokio_util::Compat<T>>;`

**Comment by Majored** [comment]
@mineichen I saw a notification about your suggestion to provide type aliases which I quite like. Unsure if there was a specific reason for your deletion but from just playing around with it myself with `with_tokio()` methods, it's actually surprisingly ergonomic.

I've written up a short explainer: https://github.com/Majored/rs-async-zip/blob/846ec137907d70f96f93d08c96277f0ec7b788d0/src/tokio/mod.rs#L6

I'm happy to settle on this unless anyone else has any suggestions.

**Comment by jplatte** [comment]
Makes sense, except the naming might be a bit confusing? I'd expect a `with_foo` constructor to always take at least one argument. Maybe `for_tokio` would be less confusing? 🤷

**Comment by Majored** [comment]
They do take the reader as a single argument but I assume you're mentioning this since the module documentation links are slightly confusing? Not sure there's anything we can do about that. Maybe omitting the parentheses helps but to me, it just looks a bit aesthetically odd without them.

**Comment by mineichen** [comment]
> anyone else has any suggestions

Sorry for the confusion. I wasn't sure if I misinterpreted your post and just suggesting what you meant already...
